enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,286 @@
+become active on my cluster without having to periodically check the Insights
+tab. I want to receive notifications through my preferred channel (e.g., Slack,
+email) configured via Alertmanager when a recommendation becomes active, and
+another one when it becomes inactive again.
+
+### Why should this be implemented in Insights Operator?
+
+The Insights Operator is already fetching the relevant data from the Smart Proxy,
+and managing Prometheus metrics based on this data. Therefore, it makes sense
+for additional information about recommendations to be implemented alongside
+what already exists.
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+
+Please note that IO refers to the Insights Operator in this context.
+
+1. IO is enabled and periodically uploads archives to the Ingress service, which
+   are then processed by the CCX data pipeline. (This is the existing behavior,
+   which will not have to be modified in any way.)
+2. After a successful archive upload, IO waits for the report to become
+   available, fetches it from the Smart Proxy and processes it. (This is an
+   extension of the current behavior, where we only look at the number of
+   active recommendations per severity level.)
+3. Prometheus metrics indicating the status of recommendations being active are
+   set accordingly. Only metrics for active recommendations are reported during
+   the metric collection. If a recommendation is no longer active, the metric
+   will not be collected, and the in-cluster Prometheus is expected to delete
+   it, which in turns should cancel the corresponding alert.
+4. The metric will have labels containing the ID, severity (based on the total
+   risk property), a human-readable name, and a link to a detailed description
+   of the recommendation and its remediation steps (Insights Advisor URL
+   generated based on the cluster ID and rule ID by IO) and its value will be
+   0/1 to indicate if the recommendation is inactive/active respectively.
+5. There will be a generic/templated info-level alert present on the cluster,
+   which will present the user with brief information about the hitting
+   recommendation (its human-readable name and severity),
+   and the link for details and remediation steps.
+
+It is worth pointing out that if any key part of the Prometheus stack is not
+running as expected on the cluster, then this feature will not work. Prometheus
+manages both the metrics and the alerts, so without it, no alerts can be
+generated by IO.
+
+An example of what the Insights recommendation metric may look like:
+```text
+insights_recommendation_active{rule_id="ccx_rules_ocp.external.rules.empty_prometheus_db_volume.report",error_key="PROMETHEUS_DB_VOLUME_IS_EMPTY",severity="moderate",description="Prometheus metrics data will be lost when the Prometheus pod is restarted or recreated",info_url="http://example.com/"} 1646416494
```

Comment:
`description` here is a longer string than I am used to seeing in Prom metrics.  And it doesn't seem to be called out in your earlier:

>  ...will have labels containing the ID, severity (based on the total risk property), a human-readable name, and a link to a detailed description of the recommendation and its remediation steps (Insights Advisor URL...)

You might also want to consider `rule_name` or just `name` or some such instead of `error_key`.  Insights rules might not even be errors, right?

Reply:
The `rule_name` and `error_key` are field names that we use throughout the Insights stack, so those are not just some arbitrarily chosen label names.

Description is the name of the field that contains the human-readable name mentioned in the quoted paragraph, and its length was one of the concerns I had initially, but after some discussions, we came to the conclusion that it should not be an issue.